### PR TITLE
Publish Daniel Hlavacek's interview

### PR DESCRIPTION
**Pull request by Julievrz**
Contributor interview 4

It would be nice to publish it on April 7

### CODE REVIEW

**Review by atomiix — APPROVED** [review]

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+**Hi Daniel, can you tell us more about yourself?**
+
+Hi, I’m Daniel. I’m 28. I live in Pardubice, Czech Republic.
+
+70% of my job is to take care of my family’s online stores. The rest of the time, I work as a freelancer. I have about 10 other clients for whom I’ve built online stores. I won’t share my website here, as I’m not taking on any new clients at the moment. My hands are full! :relaxed:
+
+My family has been selling kitchen appliances online since 2006. Our main website is [Trendo.cz](https://www.trendo.cz/). My role in the family business is to take care of our stores, product catalog, network, hardware, and everything related.
+
+I have always loved computers: building and repairing laptops, making websites, setting up networks, etc. My ecommerce journey started in 2008. I was about 15 years old when I started working for my family business - managing products and content on our website. Gradually, I took on more and more responsibilities.
```
Comment (Hlavtox) [inline news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md]: PLS change:
`building and repairing laptops`
`building them, repairing laptops`

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+I learned so much by contributing to the open-source project. I was a beginner in PHP back then. I could write some basic things and mostly submitted one-line pull requests. After 5 years, I think I can do almost anything and make complex pull requests on stuff I was afraid to touch, such as the faceted search and others.
+
+**You also take part in the translation of PrestaShop into Czech. What motivates you?**
+
+Translations needed to be improved. There weren’t many active Czech translators and I couldn’t get in contact with any of them. I studied translation and interpretation in high school, so I thought I’d give it a hand! It’s nice to see that PrestaShop is now fully available in Czech. 
+
+**Do you have any advice for first-time contributors?**
+
+First of all, I think the process could be improved. It would be nice to have more people approving and checking pull requests. Right now, if I submit a pull request, it can take several days before I get an answer. Then there are a lot of back and forths and different people taking turns. In the end, it can take a long time for a pull request to be merged and that's a bit discouraging. I think it’s a governance issue.
```
Comment (NeOMakinG) [inline news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md:62]: Looks like this is not direct advice to first-time contributors but more like a pain point he encounter while contributing, advice would have been written like: "Merging your pull request will take a lot of time, don't lose patience"

Comment (NeOMakinG) [reply news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md:62]: Ok, I will read it entirely 😆

**Review by NeOMakinG — APPROVED** [review]
Great interview, you're welcome @Hlavtox, all the thanks go to you!

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md (hunk 2 above)
Comment (Julievrz) [reply news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md:62]: 😄

**Comment by ttoine** [comment]
maybe we could ask Patrick Coffre to review it? that could a good onboarding for him?

**Comment by Julievrz** [comment]
> maybe we could ask Patrick Coffre to review it? that could a good onboarding for him?

Of course, he can review it as long as we don't change completely what Daniel said. Daniel has already reviewed it. ☺️ 

I'd like to publish it on April 7 if possible. 😇

**Comment by ttoine** [comment]
ping @pcoffre
let's review that together so I can show you how to do it. and we must get it done asap

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ - community
+ - contribution
+---
+Have you ever wondered who was hiding behind a GitHub username? It’s time to find out! Contributor interviews are a great way of getting to know the people who participate in the PrestaShop project. This month, meet Daniel Hlaváček, aka [Hlavtox](https://github.com/Hlavtox).
+
+<img style="border: 1px solid #CCC; float: left; margin: 0 1em 1em 0;" width="517" height="388" src="/assets/images/2022/04/daniel.jpg">
+
+Daniel spends most of his time managing his family business’ online stores. The rest of the time, he works as a freelance developer for 10 clients, whose stores are all based on PrestaShop.
+
+Despite his busy schedule, Daniel also takes time to contribute to the PrestaShop open-source project, with more than [300 contributions](https://contributors.prestashop.com/).
```
Comment (coffrep) [inline news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md]:
```suggestion
Despite his busy schedule, Daniel also takes time to contribute to the PrestaShop open source project, with more than [300 contributions](https://contributors.prestashop.com/). 
```

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+**When did you get involved in contributing to the PrestaShop project?** 
+
+We launched our first PrestaShop store in 2014 on the 1.6 version. Back then, I only reported some bugs on Forge[^1] .
+
+In 2018, PrestaShop [moved issues to Github](https://build.prestashop.com/news/moving-issues-to-github/), and around that time, we migrated to the 1.7 version. During the migration to 1.7, I encountered many bugs, so I reported them on Github. When I realized that the PrestaShop team couldn’t fix them all, I motivated myself and decided to contribute. So, I stopped complaining and took action.
+
+[^1]: <em> Former web-based collaborative software platform based on Jira. </em>
+
+**As a freelancer, why do you contribute to the open-source project?**
```
Comment (coffrep) [inline news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md]:
```suggestion
**As a freelancer, why do you contribute to the open source project?**
```

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+Honestly? So I have less work when upgrading my stores. The more things I fix, the less work will everybody have when upgrading. I also like the feeling of doing something good: new merchants who create a PrestaShop store will get much better software than the one I started with.
+
+Also, if something is implemented in the core, all modules will count on it and automatically include it in their data. For example, the MPN field: I added it to my PrestaShop build, but all other modules needed to be modified: XML export to Google, marketing modules, etc.
```
Comment (coffrep) [inline news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md]:
```suggestion
Also, if something is implemented in the core, all modules will count on it and automatically include it in their data. For example, the MPN field: I added it to my PrestaShop build. But all other modules needed to be modified: XML export to Google, marketing modules, etc.
```
Bonjour Julie, sur ce passage j'ai rajouté un point (".") et une nouvelle phrase commençant par "But all other" pour qu'il n'y ai pas 2 deux points (":") dans la même phrase. On connecte sur Slack si tu as une question.

File: news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+I learned so much by contributing to the open-source project. I was a beginner in PHP back then. I could write some basic things and mostly submitted one-line pull requests. After 5 years, I think I can do almost anything and make complex pull requests on stuff I was afraid to touch, such as the faceted search and others.
```
Comment (coffrep) [inline news/_posts/2022-04-07-contributor-interview-daniel-hlavacek.md]:
```suggestion
I learned so much by contributing to the open source project. I was a beginner in PHP back then. I could write some basic things and mostly submitted one-line pull requests. After 5 years, I think I can do almost anything and make complex pull requests on stuff I was afraid to touch, such as the faceted search and others.
```

**Review by coffrep — CHANGES_REQUESTED** [review]

**Comment by ttoine** [comment]
FYI
https://opensource.com/submission-style-guide#style

open source, not open-source.

**Comment by Julievrz** [comment]
> FYI https://opensource.com/submission-style-guide#style
> 
> open source, not open-source.

Hi @ttoine 

Here "open source" is a compound adjective preceding a noun so it should be hyphenated. But I agree that "open source" should not be hyphenated when it is a noun. This rule appears in the Content style guide and the decision was based on many style guides, such as the Microsoft one:

Microsoft: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/open-source#:~:text=Hyphenate%20open%2Dsource%20as%20an,abbreviation%20for%20open%2Dsource%20software.

**Comment by ttoine** [comment]
Writing "open-source" is not a standard. Microsoft is quite a recent actor in open source and should not be trusted as a reference on the topic. I think that they write it this way because this how you find on Wikipedia for years, but I don't really understand why.

You will see on [opensource.google](opensource.google) or [https://opensource.apple.com/](https://opensource.apple.com/) that there is no "-", no matter how it is used.

Same if we look at historic organisations that are pioneer of open source, [https://www.eclipse.org/](https://www.eclipse.org/), [https://linuxfoundation.org/resources/open-source-guides/](https://linuxfoundation.org/resources/open-source-guides/) or [https://opensource.org/history](https://opensource.org/history)

It is rather a qualifier than an adjective, even if placed before software.
 
Patrick and I started to publish content about open source software in the 2000s and both have never write that way. So, indeed, we made suggestions based on our experience. The point is not to be in conflict with you or to tell that you are wrong...

So, the question is: do we want to follow Microsoft's recommandations, or do we want to use open source historic actors recommandations...

**Comment by Julievrz** [comment]
> Writing "open-source" is not a standard. Microsoft is quite a recent actor in open source and should not be trusted as a reference on the topic. I think that they write it this way because this how you find on Wikipedia for years, but I don't really understand why.
> 
> You will see on [opensource.google](opensource.google) or https://opensource.apple.com/ that there is no "-", no matter how it is used.
> 
> Same if we look at historic organisations that are pioneer of open source, https://www.eclipse.org/, https://linuxfoundation.org/resources/open-source-guides/ or https://opensource.org/history
> 
> It is rather a qualifier than an adjective, even if placed before software.
> 
> Patrick and I started to publish content about open source software in the 2000s and both have never write that way. So, indeed, we made suggestions based on our experience. The point is not to be in conflict with you or to tell that you are wrong...
> 
> So, the question is: do we want to follow Microsoft's recommandations, or do we want to use open source historic actors recommandations...

I'm not upset at all, on the contrary. ☺️ These resources do indeed prove your point and I will modify the Content style guide accordingly. I am not at all against changing the rules of the Content style guide when it is justified. So thanks for sharing your arguments and resources.

This is however a tricky question as the Open Source Initiative hyphenates open source when it's an adjective. https://opensource.org/osd 🤯. Let's just take a decision and stick to it. ;-)

**Comment by ttoine** [comment]
yes, a very tricky question. I asked both OSI and opensource.org on Twitter about what they recommend. let see if they take the time to answer.

**Comment by eternoendless** [comment]
Thank you @Julievrz

**Pull request merged**